source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -236,6 +239,73 @@ class nifgenDriverApiTest : public ::testing::Test {
     EXPECT_EQ(kfgenDriverApiSuccess, import_response.status());
   }
 
+  void configure_output_mode(const char* channel_name, fgen::OutputMode output_mode)
+  {
+    ::grpc::ClientContext context;
+    fgen::ConfigureOutputModeRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_output_mode(output_mode);
+    fgen::ConfigureOutputModeResponse response;
+
+    ::grpc::Status status = GetStub()->ConfigureOutputMode(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  }
+
+  ViInt32 create_waveform_f64(const char* channel_name, ViInt32 waveform_size, ViReal64 waveform_data_array[])
+  {
+    ::grpc::ClientContext context;
+    fgen::CreateWaveformF64Request request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_channel_name(channel_name);
+    for (int i = 0; i < waveform_size; i++) {
+      request.add_waveform_data_array(waveform_data_array[i]);
+    }
+    fgen::CreateWaveformF64Response response;
+
+    ::grpc::Status status = GetStub()->CreateWaveformF64(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+    return response.waveform_handle();
+  }
+
+  ViInt32 create_sine_waveform(const char* channel_name)
+  {
+    ViInt32 waveform_size = 64;
+    ViReal64 sine_waveform_data[64];
+    double x;
+    for (int i = 0; i < waveform_size; i++) {
+      x = ((double)i / waveform_size) * 2 * M_PI;
+      sine_waveform_data[i] = std::sin(x);
+    }
+    ViInt32 sine_waveform_handle = create_waveform_f64(channel_name, waveform_size, sine_waveform_data);
+    return sine_waveform_handle;
+  }
+
+  int create_advanced_arb_sequence(ViInt32 sequence_length, ViInt32 waveform_handles_array[], ViInt32 loop_counts_array[], ViInt32 marker_location_array[])
+  {
+    ::grpc::ClientContext context;
+    fgen::CreateAdvancedArbSequenceRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    for (int i = 0; i < sequence_length; i++) {
+      request.add_waveform_handles_array(waveform_handles_array[i]);
+    }
+    for (int i = 0; i < sequence_length; i++) {
+      request.add_loop_counts_array(loop_counts_array[i]);
+    }
+    for (int i = 0; i < sequence_length; i++) {
+      request.add_marker_location_array(marker_location_array[i]);
+    }
+    fgen::CreateAdvancedArbSequenceResponse response;
+
+    ::grpc::Status status = GetStub()->CreateAdvancedArbSequence(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    return response.status();
```

Comment:
Do you think we can check the number of elements in the output array (coerced array) will add any  value in checking our gRPC implementation logic of the API ?

Reply:
@ganeshnie , I also initially thought of asserting the size of returned array. But I remembered that for one of the previous drivers, owners had told that in system tests, we should not be adding assert statements for the data returned. And they had told that, if we want to test the data, we have to write unit tests for it. But till now, we have only written unit tests for fake tests. But I can attempt to create a new test class for Fgen unit tests. Shall I go ahead and create a unit tests class for fgen?

Reply:
Discussed offline that it wont add much value. Hence resolving the comment